docs/zkapps/tutorials/09-recursion.mdx

```diff
@@ -56,86 +56,310 @@ You build recursive zkApps with [ZkProgram](/zkapps/o1js-reference/modules/Exper
+26      },
+27    },
+28  },
+29 });
 ```
 
 Use recursion to combine two proofs:
 
 ```typescript
-  add: {
-    privateInputs: [ SelfProof, SelfProof ],
-
-    method(
-      newState: Field,
-      earlierProof1: SelfProof<Field>,
-      earlierProof2: SelfProof<Field>,
-    ) {
-      earlierProof1.verify();
-      earlierProof2.verify();    	
-      newState.assertEquals(earlierProof1.publicInput.add(earlierProof2.publicInput));
-    },
-  },
+28
+29    add: {
+30      privateInputs: [SelfProof, SelfProof],
+31
+32      method(
+33        newState: Field,
+34        earlierProof1: SelfProof<Field, void>,
+35        earlierProof2: SelfProof<Field, void>
+36      ) {
+37        earlierProof1.verify();
+38        earlierProof2.verify();
+39        newState.assertEquals(
+40          earlierProof1.publicInput.add(earlierProof2.publicInput)
+41        );
+42      },
+43    },
+44  },
+45 });
 ```
 
-To use ZkProgram, compile it and then call methods on it:
+## Interact with a smart contract
+
+Next, write a script that interacts with your smart contract. As before, the complete [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/01-hello-world/src/main.ts) example file is provided. Follow these steps to build the `main.ts` file so you can interact with the smart contract.
+
+### Imports
+
+For this tutorial, the `import` statement brings in items from `o1js` that you use to interact with your smart contract. 
+
+1. Copy the following lines from [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/01-hello-world/src/main.ts) example file into the `src/main.ts` file:
+
+```ts src/main.ts
+1 import { Add } from './Add.js';
+2 import { Field, verify } from 'o1js';
+```
+
+### Call Methods
+
+Now we will add methods to call ZkProgram:
 
 ```typescript
-  console.log('compiling...');
+3
+4 async function main() {
+5   console.log('compiling...');
+6 
+7   const { verificationKey } = await Add.compile();
+8 
+9   console.log('making proof 0')
+10  
+11   const proof0 = await Add.init(Field(0));
+12  
+13   console.log('making proof 1')
+14  
+15   const proof1 = await Add.addNumber(Field(4), proof0, Field(4));
+16  
+17   console.log('making proof 2')
+18 
+19   const proof2 = await Add.add(Field(4), proof1, proof0);
+20 
+21   console.log('verifying proof 2');
+22   console.log('proof 2 data', proof2.publicInput.toString());
+23  
+24   const ok = await verify(proof2.toJSON(), verificationKey);
+25   console.log('ok', ok);
+26 }
+27 
+28 main();
+```
 
-  const { verificationKey } = await Add.compile();
+Verification of the proof can occur off-chain using the `verify()` method. This is useful for applications where you want to prove something to an off-chain entity.
 
-  console.log('making proof 0')
+### Build and run the smart contract
 
-  const proof0 = await Add.init(Field(0));
+Now that the Square smart contract is complete, these commands run your project as a local blockchain.
```

Comment:
what is the contract we want to build? Square? (be sure all the file names and example files match up throughout)

Reply:
Oh Sorry, I missed this one.
Will proof read it once again.

According to existing tutorial and the examples, we are building three files mainly:
- Add.ts
- Rollup.ts
- Vote.ts